### PR TITLE
Delete old TensorFlow Bazel code, docs, and flags.

### PR DESCRIPTION
**Pull request by ScottTodd**
Follow-up to the work tracked on https://github.com/openxla/iree/issues/13037.

### CODE REVIEW

**Review by stellaraccident — APPROVED** [review]

File: build_tools/scripts/integrate/README.md
Comment (ScottTodd) [inline build_tools/scripts/integrate/README.md:1]: FYI @MaheshRavishankar - I'm removing the "Sync everything to a Google/TensorFlow commit" part of these docs. A few other parts of this doc could use a review pass from someone more familiar with the process.

File: build_tools/bazel/iree.bazelrc [hunk 1]
```diff
@@ -277,7 +273,6 @@ build:remote_cache_bazel_ci --config=_remote_cache_base
 # included for clarity and so that this reference is automatically updated by
 # manage_images.py
 build:remote_cache_bazel_ci --host_platform_remote_properties_override='properties:{name:"cache-silo-key" value:"gcr.io/iree-oss/swiftshader-bleeding-edge@sha256:b3cce73eb9f41d67981bc8f00e98fd66fe3487caec1bcbf38d4039dcc61e499d"}'
-build:remote_cache_bazel_tf_ci --host_platform_remote_properties_override='properties:{name:"cache-silo-key" value:"gcr.io/iree-oss/frontends-swiftshader@sha256:da14cc93637d3bfad469a670d4d7a49982df5d107b775331965e3bacb981d4cf"}'
```
Comment (ScottTodd) [inline build_tools/bazel/iree.bazelrc:280]: @GMNGeoffrey I think this was only used by a Buildkite pipeline that is no longer in use. Seemed safe to remove. WDYT?

Comment (GMNGeoffrey) [reply build_tools/bazel/iree.bazelrc:280]: I still see [usages](https://github.com/search?q=remote_cache_bazel_tf_ci+repo%3Aopenxla%2Firee&type=code) in the repo. Those either need to be updated or deleted (and then usages of them deleted, etc.). @pzread would know best about the current status of buildkite benchmark pipelines and whether the same scripts are being used for running locally as well.

File: build_tools/scripts/integrate/README.md
Comment (GMNGeoffrey) [reply build_tools/scripts/integrate/README.md:1]: I think the Google commit still actually provides value in that it indicates a commit that has received some amount of testing. For now, it also is necessary to occasionally use these commits for our internal integrations.

File: build_tools/bazel/iree.bazelrc (hunk 1 above)
Comment (GMNGeoffrey) [reply build_tools/bazel/iree.bazelrc:280]: Oh, I see you're doing that with https://github.com/openxla/iree/pull/13435. Then yeah this LGTM assuming that is submitted first

Comment (ScottTodd) [reply build_tools/bazel/iree.bazelrc:280]: Yep. I can shuffle code between PRs as needed (some of the changes in that other PR may be more controversial).

File: build_tools/scripts/integrate/README.md
Comment (ScottTodd) [reply build_tools/scripts/integrate/README.md:1]: I can keep some part of the docs, but sections like
```
sed -i "s/^TENSORFLOW_COMMIT = .*$/TENSORFLOW_COMMIT = \"$TF_COMMIT\"/" integrations/tensorflow/WORKSPACE
```

are outdated (`integrations/tensorflow/WORKSPACE` no longer exists)

**Pull request merged**